src/http/retry_strategy.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#ifdef SEASTAR_MODULE
+module;
+#endif
+
+#include <chrono>
+#include <coroutine>
+#include <gnutls/gnutls.h>
+
+#ifdef SEASTAR_MODULE
+module seastar;
+#else
+#include <seastar/http/exception.hh>
+#include <seastar/http/retry_strategy.hh>
+#include <seastar/util/short_streams.hh>
+
+#endif
+
+using namespace std::chrono_literals;
+
+namespace seastar {
+extern logger http_log;
+
+namespace http::experimental {
+logger rs_logger("default_http_retry_strategy");
+
+extern bool is_retryable_exception(std::exception_ptr ex);
+
+default_retry_strategy::default_retry_strategy(unsigned max_retries, unsigned scale_factor)
+    : _max_retries(max_retries), _scale_factor(scale_factor) {
+}
+
+default_retry_strategy::default_retry_strategy()
+    : default_retry_strategy(2, 25) /* Default overall retry time to fail is 50ms (0ms)+((1<<1)*25)ms) */ {
```

Comment:
Why 2? Or does _max_retries include the initial attempt as well?

Reply:
The way I see it -- try, retry, retry, ... so the first attempt is a try, not retry